pkg/api/customization/cred/store.go

```diff
@@ -65,3 +76,43 @@ func Validator(request *types.APIContext, schema *types.Schema, data map[string]
 
 	return nil
 }
+
+type Store struct {
+	types.Store
+	NodePoolLister     v3.NodePoolLister
+	NodeTemplateLister v3.NodeTemplateLister
+}
+
+func (s *Store) Delete(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
+	nodeTemplates, err := s.NodeTemplateLister.List("", labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+	if len(nodeTemplates) > 0 {
+		toCheck := map[string]bool{}
+		for _, template := range nodeTemplates {
+			if template.Spec.CloudCredentialName != id {
+				continue
+			}
+			toCheck[template.Name] = true
+		}
+		if len(toCheck) > 0 {
+			pools, err := s.NodePoolLister.List("", labels.Everything())
```

Comment:
we should disallow removing the CC even if the node template is not in use by a node pool. Because once CC is removed, the node template can't really be used for the node pool creation, and should be cleared along with CC.